epix/run_epix.py

```diff
@@ -107,13 +107,26 @@ def main(args, return_df=False, return_wfsim_instructions=False, strax=False):
         print('Generating photons and electrons for events')
     # Generate quanta:
     if len(result) > 0:
-        photons, electrons, excitons = epix.quanta_from_NEST(epix.awkward_to_flat_numpy(result['ed']),
-                                                             epix.awkward_to_flat_numpy(result['nestid']),
-                                                             epix.awkward_to_flat_numpy(result['e_field']),
-                                                             epix.awkward_to_flat_numpy(result['A']),
-                                                             epix.awkward_to_flat_numpy(result['Z']),
-                                                             epix.awkward_to_flat_numpy(result['create_S2']),
-                                                             density=epix.awkward_to_flat_numpy(result['xe_density']))
+        if not ('yield' in args.keys()): 
+            print("No yield is provided! Forcing nest")
+            args['yield']="nest"
+        if args['yield'].lower()=="nest":
+            photons, electrons, excitons = epix.quanta_from_NEST(epix.awkward_to_flat_numpy(result['ed']),
+                                                                 epix.awkward_to_flat_numpy(result['nestid']),
+                                                                 epix.awkward_to_flat_numpy(result['e_field']),
+                                                                 epix.awkward_to_flat_numpy(result['A']),
+                                                                 epix.awkward_to_flat_numpy(result['Z']),
+                                                                 epix.awkward_to_flat_numpy(result['create_S2']),
+                                                                 density=epix.awkward_to_flat_numpy(result['xe_density']))
+        elif args['yield'].lower()=="bbf":
+            bbfyields=epix.BBF_quanta_generator()
+            photons, electrons, excitons = bbfyields.get_quanta_vectorized(
+                                energy=epix.awkward_to_flat_numpy(result['ed']),
+                                interaction=epix.awkward_to_flat_numpy(result['nestid']),
+                                field=epix.awkward_to_flat_numpy(result['e_field'])
+                                )
+        else:
+            raise RuntimeError("Unknown yield model: ", args['yields'])
```

Comment:
Hi @terliuk thanks for starting to implement this to epix. I think it would keep the code a bit cleaner if we set up a single quanta generation class containing both the NEST and BBF option. In run_epix.py we would then just call this one class.

Reply:
I suggest that we open a ticket and refactor it later, as I do not think that now making code nice is a priority, to be honest.